src/main/java/com/puppycrawl/tools/checkstyle/JavaParser.java

```diff
@@ -89,22 +92,64 @@ public static DetailAST parse(FileContents contents)
         final CommonTokenStream tokenStream = new CommonTokenStream(lexer);
         final JavaLanguageParser parser =
                 new JavaLanguageParser(tokenStream, JavaLanguageParser.CLEAR_DFA_LIMIT);
-        parser.setErrorHandler(new CheckstyleParserErrorStrategy());
+        JavaLanguageParser.CompilationUnitContext compilationUnit;
+
+        try {
+            compilationUnit = parseWithSLLPredictionMode(parser);
+        }
+        catch (ParseCancellationException e) {
```

Comment:
```suggestion
        catch (ParseCancellationException ignored) {
```
`ex` or more precise `ignored` is the pattern imho. I expect the build to fail as imho we have a pattern avoiding one character names like `e`.